### ISSUE TITLE
Distinct operation on UUID fails

### ISSUE THREAD

**Issue by 16pierre** [prestodb/presto#23554]
## Your Environment

Local.

## Expected Behavior

Group by operations on UUIDs should work.

## Current Behavior

Group by on UUID throws:
```
java.lang.UnsupportedOperationException: com.facebook.presto.common.block.Int128ArrayBlockBuilder
	at com.facebook.presto.common.block.Block.compareTo(Block.java:188)
	at com.facebook.presto.type.UuidOperators$UuidDistinctFromOperator.isDistinctFrom(UuidOperators.java:202)
	at com.facebook.presto.$gen.PagesHashStrategy_20240829_181240_24.positionNotDistinctFromRow(Unknown Source)
	at com.facebook.presto.operator.MultiChannelGroupByHash.positionNotDistinctFromCurrentRow(MultiChannelGroupByHash.java:465)
	at com.facebook.presto.operator.MultiChannelGroupByHash.putIfAbsent(MultiChannelGroupByHash.java:306)
	at com.facebook.presto.operator.MultiChannelGroupByHash.putIfAbsent(MultiChannelGroupByHash.java:296)
	at com.facebook.presto.operator.MultiChannelGroupByHash.access$200(MultiChannelGroupByHash.java:55)
	at com.facebook.presto.operator.MultiChannelGroupByHash$AddNonDictionaryPageWork.process(MultiChannelGroupByHash.java:615)
	at com.facebook.presto.operator.HashAggregationOperator.addInput(HashAggregationOperator.java:423)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:451)
	at com.facebook.presto.operator.Driver.lambda$processFor$10(Driver.java:324)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:750)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:317)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1079)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:165)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:621)
	at com.facebook.presto.$gen.Presto_0_288_15f14bb____20240829_181132_1.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

Same issue was reported on Trino: https://github.com/trinodb/trino/issues/1170

## Possible Solution

Was fixed in Trino with a pretty simple one-liner apparently: https://github.com/trinodb/trino/pull/1180

## Steps to Reproduce

```
select distinct u 
from (values (uuid'be0b0518-35a1-4d10-b7f1-1b61355fa741'),
             (uuid'be0b0518-35a1-4d10-b7f1-1b61355fa741')
) as t(u)
```

## Context

I perform `GROUP BY` on UUIDs. 
To workaround this bug I need to cast UUIDs to strings, not great for performance.

**Comment by 16pierre** [comment]
Happy to give it a shot

**Comment by itsarijitray** [comment]
@16pierre Are you working on this? If not, I can give it a try

**Comment by 16pierre** [comment]
Sure feel free to, haven't started yet

**Comment by itsarijitray** [comment]
Sure, thanks!

**Comment by abhibongale** [comment]
@16pierre @itsarijitray can I give a try?

**Comment by ZacBlanco** [comment]
This has been fixed by #23732
